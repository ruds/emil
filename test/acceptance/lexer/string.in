"this is a string"
"this is a string \   \ with a gap"
"this string splits lines \
   \across a gap."
"this string has a \  \ \"gap\" and some \aescapesðŸ˜ƒ"
""
"No line breaks allowed without a gap
(*xx*)
""""""
"""This is
a triple string
with \nescapes like \U01f603"""
"""No gaps \ \ allowed"""
(*xx*)
r"(raw strings \  \ don't \" escape characters \U01f603)"
r"==(if I have delimiters I can put )" in the
middle of my string)=="
R"(The capital R also works, but R"(raw strings don't nest)"
R"(This string never terminates
