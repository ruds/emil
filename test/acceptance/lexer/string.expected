0001 STRING 74686973206973206120737472696E67 "this is a string"
0002 STRING 74686973206973206120737472696E67202077697468206120676170 "this is a string \   \ with a gap"
0003 STRING 7468697320737472696E672073706C697473206C696E6573206163726F73732061206761702E "this string splits lines \
   \across a gap."
0005 STRING 7468697320737472696E672068617320612020226761702220616E6420736F6D65200765736361706573F09F9883 "this string has a \  \ \"gap\" and some \aescapesðŸ˜ƒ"
0006 STRING ""
0007 ERROR
0009 STRING """"""
0010 STRING 546869732069730A6120747269706C6520737472696E670A77697468200A65736361706573206C696B6520F09F9883 """This is
a triple string
with \nescapes like \U01f603"""
0013 ERROR
0015 STRING 72617720737472696E6773205C20205C20646F6E2774205C22206573636170652063686172616374657273205C55303166363033 r"(raw strings \  \ don't \" escape characters \U01f603)"
0016 STRING 6966204920686176652064656C696D697465727320492063616E2070757420292220696E207468650A6D6964646C65206F66206D7920737472696E67 r"==(if I have delimiters I can put )" in the
middle of my string)=="
0018 STRING 546865206361706974616C205220616C736F20776F726B732C206275742052222872617720737472696E677320646F6E2774206E657374 R"(The capital R also works, but R"(raw strings don't nest)"
0019 ERROR
0020 END_OF_FILE
